97.md

```diff
@@ -0,0 +1,251 @@
+NIP-97
+=======
+
+Attachments
+-----------
+
+`final` `optional` `author:ondra-novak`
+
+This NIP solves the problem of sharing and distributing binary content over NOSTR networks such as media - images, short videos music, or in general any files up to a certain size
+
+Key Takeaways
+-------------
+
+* Defines how binary content (images, media, files) is managed by NOSTR network
+* Allows sending private binary content in direct messages (encrypted)
+* Defines new commands at the protocol level ("ATTACH","FETCH")
+* Defines a new tag "attachment"
+* Uses the binary messages in the websocket connection
+
+
+New tag "attachment"
+-------------
+The proposal introduces a new tag  "attachment" with the following format
+
+```
+["attachment","hash_hex","size_in_bytes","mime_type","features1", "feature2", ....]
+```
+
+* **hash_hex** - hexadecimal representation of SHA256 hash of the attached binary file. This string also serves as **attachment-id**
+* **size_in_bytes** - size of the attachment in bytes
+* **mime_type** - mime type of the attachment 
+* **features...** - optional fields with various features in format `<field>=<value>`
+    * **blurhash=** - contains NIP-94 blurhash
+    * **purpose=** - purpose of the attachment in context of current note. This can help client
+to select optimal rendering. Suggested purposes
+        * **thumbnail** - thubnail image of a following video
+        * **story** - all attachments marked by this purpose are played as a story
+        * **inline** - attachment is rendered in text, referenced by `#[<index>]`. This purpose hides this attachment from the list of attachments in note's detail.
+
+```
+["attachment","12887...","64000","image/jpeg","blurhash=qwdjq3...", "purpose=thumbnail"]
+["attachment","874788...","2258147","video/mpeg4", "purpose=story"]
+["attachment","ae3c758...","2158799","video/mpeg4", "purpose=story"]
+["attachment","788eq78...","254123","image/jpeg", "purpose=story"]
+```
+        
+
+* One event can have more than one of these tags, then it carries more attachments.
+* Each relay defines the maximum attachment size and also the maximum number of attachments within an event
+* Attachment can be attached to any "kind". Attachments for `kind:4` are always encrypted
+
+The presence of the `attachment` attribute only informs the client that the relay should also have its own attachment content available in binary form. The client must download the attachments from the same relay to display them. See below
```

Comment:
SHOULD or MUST? If relays have the event, they MUST have the attachment as well. Otherwise, it gets a little messy on where to find it. 

Relays that don't support this NIP will all have the event, but not the attachment. Thus, I don't think there is a good solution for MUST. Clients must always assume the relays might not have it and figure out a way to discover which relay has the attachment.

Reply:
SHOULD as your client must be fault tolerant. Similarly, the image you point to via the URL may be missing

If the relay does not have an attachment, the client is under no obligation to look for it elsewhere. He just doesn't have it, it's not available. Broken image icon.